autopush/truss/handler.py

```diff
@@ -0,0 +1,234 @@
+import time
+import uuid
+import json
+
+from boto.dynamodb2.exceptions import ItemNotFound
+from twisted.internet.threads import deferToThread
+from twisted.internet.defer import Deferred
+
+from autopush.base import (
+    BaseWebHandler,
+    threaded_validate,
+)
+from autopush.db import (
+    generate_last_connect,
+    hasher,
+)
+from autopush.utils import generate_hash, ms_time
+from autopush.truss.validation import (
+    TrussSchema,
+)
+
+
+class RegistrationHandler(BaseWebHandler):
+    cors_methods = "POST,PUT,DELETE"
+
+    #############################################################
+    #                    Cyclone HTTP Methods
+    #############################################################
+    @threaded_validate(TrussSchema)
+    def post(self, truss_data):
+        """HTTP POST
+
+        Endpoint generation and optionally router type/data registration.
+
+        :param truss_data: Dictionary of validated values.
```

Comment:
the validator actually *doesn't* pass the validated dict as this argument, it only sets it as self.valid_input and passes along the args/kwargs from Cyclone. So RegistrationHandler is actually broken right now (Cyclone should still be calling it with 4 arguments)

Which reminds me, I would actually *like* the validator to work like that instead of setting self.valid_input. feels cleaner to me and AFAICT we're always passing all of the cyclone args through the validation step anyway.. passing them along unvalidated is confusing.

but let's just fix the args here for now and think about the validator change later